### PR TITLE
sql-server command documentation

### PR DESCRIPTION
**Pull request by bheni**
```
NAME
	dolt sql-server - Start a MySQL-compatible server.

SYNOPSIS
	dolt sql-server --config <file>
	dolt sql-server [-H <host>] [-P <port>] [-u <user>] [-p <password>] [-t <timeout>] [-l <loglevel>] [--multi-db-dir <directory>] [-r]

DESCRIPTION
	By default, starts a MySQL-compatible server which allows only one user connection ata time to the dolt repository in the current directory. Any edits made through this server will be automatically reflected
	in the working set.  This behavior can be modified using a yaml configuration file passed to the server via --config <file>, or by using the supported switches and flags to configure the server directly on
	the command line (If --config <file> is provided all other command line arguments are ignored).

	This is an example yaml configuration file showing all supported items and their default values:


	log_level: info

	behavior:
	  read_only: false
	  autocommit: true

	user:
	  name: root
	  password: ""

	listener:
	  host: localhost
	  port: 3306
	  max_connections: 1
	  read_timeout_millis: 30000
	  write_timeout_millis: 30000

	databases: []


	SUPPORTED CONFIG FILE FIELDS:

	log_level - Level of logging provided. Options are: `trace', `debug`, `info`, `warning`, `error`, and `fatal`.

	behavior.read_only - If true database modification is disabled.

	behavior.autocommit - If true write queries will automatically alter the working set.  When working with autocommit enabled it is highly recommended that listener.max_connections be set to 1 as concurrency
	issues will arise otherwise.

	user.name - The username that connections should use for authentication.

	user.password - The password that connections should use for authentication.
	listener.host - The host address that the server will run on.  This may be `localhost` or an IPv4 or IPv6 address.

	listener.port - The port that the server should listen on.

	listener.max_connections - The number of simultaneous connections that the server will accept.

	listener.read_timeout_millis - The number of milliseconds that the server will wait for a read operation.

	listener.write_timeout_millis - The number of milliseconds that the server will wait for a write operation.

	databases - a list of dolt data repositories to make available as SQL databases. If databases is missing or empty then the working directory must be a valid dolt data repository which will be made available
	as a SQL database

	databases[i].path - A path to a dolt data repository.

	databases[i].name - The name that the database corresponding to the given path should be referenced via SQL.


	If a config file is not provided many of these settings may be configured on the command line.

OPTIONS
	--config=<file>
	  When provided configuration is taken from the yaml config file and all command line parameters are ignored.

	-H <Host address>, --host=<Host address>
	  Defines the host address that the server will run on (default `localhost`)

	-P <Port>, --port=<Port>
	  Defines the port that the server will run on (default `3306`)

	-u <User>, --user=<User>
	  Defines the server user (default `root`)

	-p <Password>, --password=<Password>
	  Defines the server password (default ``)

	-t <Connection timeout>, --timeout=<Connection timeout>
	  Defines the timeout, in seconds, used for connections
	  A value of `0` represents an infinite timeout (default `30000`)

	-r, --readonly
	  Disables modification of the database

	-l <Log level>, --loglevel=<Log level>
	  Defines the level of logging provided
	  Options are: `trace', `debug`, `info`, `warning`, `error`, `fatal` (default `info`)

	--multi-db-dir=<directory>
	  Defines a directory whose subdirectories should all be dolt data repositories accessible as independent databases.

	--no-auto-commit
	  When provided sessions will not automatically commit their changes to the working set. Anything not manually committed will be lost.
```

### CODE REVIEW

File: go/cmd/dolt/commands/sqlserver/sqlserver.go [hunk 1]
```diff
@@ -46,8 +46,46 @@ const (
 
 var sqlServerDocs = cli.CommandDocumentationContent{
 	ShortDesc: "Start a MySQL-compatible server.",
-	LongDesc:  `Start a MySQL-compatible server which can be connected to by MySQL clients.`,
+	LongDesc: "In it's default configuration, starts a MySQL-compatible server which allows only one user connection at" +
```
Comment (zachmu) [inline go/cmd/dolt/commands/sqlserver/sqlserver.go:49]: By default,

File: go/cmd/dolt/commands/sqlserver/sqlserver.go [hunk 2]
```diff
@@ -46,8 +46,46 @@ const (
...
+		"behavior.read_only - If true database modification is disabled.\n" +
+		"\n" +
+		"behavior.autocommit - If true write queries will automatically alter the working set.  When working with autocommit " +
+		"enabled it is highly recommended that listener.max_connections be set to 1 as concurrency issues will arise otherwise.\n" +
+		"\n" +
+		"user.name - The username that connections should use for authentication.\n" +
+		"\n" +
+		"user.password - The password that connections should use for authentication.\n" +
+		"\n" +
+		"listener.host - The host address that the server will run on.  This may be an `localhost` or an IPv4 or IPv6 address.\n" +
```
Comment (zachmu) [inline go/cmd/dolt/commands/sqlserver/sqlserver.go:74]: This may be `localhost`

File: go/cmd/dolt/commands/sqlserver/sqlserver.go [hunk 3]
```diff
@@ -46,8 +46,46 @@ const (
...
+		"\n" +
+		"listener.port - The port that the server should listen on.\n" +
+		"\n" +
+		"listener.max_connections - The number of simultaneous connections that the server will accept.\n" +
+		"\n" +
+		"listener.read_timeout_millis - The number of milliseconds that the server will wait for a read operation.\n" +
+		"\n" +
+		"listener.write_timeout_millis - The number of milliseconds that the server will wait for a write operation.\n" +
+		"\n" +
+		"\n" +
```
Comment (zachmu) [inline go/cmd/dolt/commands/sqlserver/sqlserver.go:84]: Should document the databases field as well

File: go/cmd/dolt/commands/sqlserver/sqlserver.go (hunk 2 above)
Comment (bheni) [reply go/cmd/dolt/commands/sqlserver/sqlserver.go:74]: updated

File: go/cmd/dolt/commands/sqlserver/sqlserver.go (hunk 1 above)
Comment (bheni) [reply go/cmd/dolt/commands/sqlserver/sqlserver.go:49]: updated

File: go/cmd/dolt/commands/sqlserver/sqlserver.go (hunk 3 above)
Comment (bheni) [reply go/cmd/dolt/commands/sqlserver/sqlserver.go:90]: done

**Pull request merged**